### PR TITLE
Weekly Support Edge Refresh

### PR DESCRIPTION
**Pull request by mkerensky**
Refreshes the edge points available to support personnel each week. Combat personnel already get their edge points back at the start of each MM battle in which they are deployed. Available support edge points are displayed on the PersonViewPanel.

### CODE REVIEW

File: MekHQ/src/mekhq/campaign/personnel/Person.java [hunk 1]
```diff
@@ -1443,6 +1446,13 @@ public void writeToXml(PrintWriter pw1, int indent) {
                         + "<edge>"
                         + String.valueOf(getOptionList("::", PilotOptions.EDGE_ADVANTAGES))
                         + "</edge>");
+            // For support personnel, write an available edge value
+            if (isSupport() || isEngineer()) {
```
Comment (sixlettervariables) [inline MekHQ/src/mekhq/campaign/personnel/Person.java]: Should this check for isAdmin etc too, like above?

File: MekHQ/src/mekhq/campaign/Campaign.java [hunk 2]
```diff
@@ -2877,6 +2877,13 @@ public void processNewDayPersonnel() {
                     u.resetPilotAndEntity();
                 }
             }
+            
+            // Reset edge points to the purchased value each week. This should only
+            // apply for support personnel - combat troops reset with each new mm game
+            if ((p.isAdmin() || p.isDoctor() || p.isEngineer() || p.isTech())
```
Comment (sixlettervariables) [inline MekHQ/src/mekhq/campaign/Campaign.java]: In Person you're checking for isSupport, but not here. Should those two places be consistent?

File: MekHQ/src/mekhq/campaign/personnel/Person.java [hunk 3]
```diff
@@ -1660,6 +1670,8 @@ public static Person generateInstanceFromXML(Node wn, Campaign c, Version versio
                     advantages = wn2.getTextContent();
                 } else if (wn2.getNodeName().equalsIgnoreCase("edge")) {
                     edge = wn2.getTextContent();
+                } else if (wn2.getNodeName().equalsIgnoreCase("edgeAvailable")) {
+                    retVal.currentEdge = Integer.parseInt(wn2.getTextContent());
```
Comment (sixlettervariables) [inline MekHQ/src/mekhq/campaign/personnel/Person.java]: Does `retVal.currentEdge` need to start as -1 and get set to edge at the end if it stays -1? I'm thinking of the case where you load a campaign and the people have not been updated yet.

File: MekHQ/src/mekhq/gui/view/PersonViewPanel.java [hunk 4]
```diff
@@ -725,6 +729,30 @@ private void fillStats() {
             gridBagConstraints.fill = GridBagConstraints.NONE;
             gridBagConstraints.anchor = GridBagConstraints.NORTHWEST;
             pnlStats.add(lblEdge2, gridBagConstraints);
+            
+            if (person.isAdmin() || person.isDoctor() || person.isTech()) {
```
Comment (sixlettervariables) [inline MekHQ/src/mekhq/gui/view/PersonViewPanel.java]: Again, should this be consistent with the other checks? (Support, engineer, etc)

File: MekHQ/src/mekhq/campaign/personnel/Person.java (hunk 1 above)
Comment (mkerensky) [reply MekHQ/src/mekhq/campaign/personnel/Person.java]: Doesn't need to. isSupport() covers astechs and medtechs not covered by isTech() and isDoctor() respectively.  Neither can use edge currently, but I didn't mind them having the value.

File: MekHQ/src/mekhq/campaign/Campaign.java (hunk 2 above)
Comment (mkerensky) [reply MekHQ/src/mekhq/campaign/Campaign.java]: See above about the differences between isSupport and isAdmin/Doctor/Tech.  In part I was experimenting with the different ways to handle this, and in the other part, I didn't want to reset edge for Astechs and Medtechs - I have a lot of Astech/Vehicle Crews and Soldier/Medtechs in my campaigns that I wanted to handle as combat personnel edgewise.    If we need consistency or if I get some compelling reasons from people who play differently I can change it.

File: MekHQ/src/mekhq/campaign/personnel/Person.java (hunk 3 above)
Comment (mkerensky) [reply MekHQ/src/mekhq/campaign/personnel/Person.java]: I'm not quite sure I understand the case you're describing.  I want to be able to save and load campaigns where currentEdge has been spent for the week, and that seems to work as I planned in testing.

File: MekHQ/src/mekhq/gui/view/PersonViewPanel.java (hunk 4 above)
Comment (mkerensky) [reply MekHQ/src/mekhq/gui/view/PersonViewPanel.java]: This is very intentional. I don't want the person view for Astechs and Medtechs to show available edge that they can't use.  The Ship's Engineer is a composite crewman that doesn't actually have a person view -  individual Vessel Crewmembers are what shows up here, and they're technically combat personnel that shouldn't show the new value either, in my mind.

**Pull request merged**